### PR TITLE
Reworked fountain, announces and teams

### PR DESCRIPTION
**Pull request by horato**

### CODE REVIEW

File: GameServer/Logic/GameObjects/Announce.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LeagueSandbox.GameServer.Logic.GameObjects
+{
+    public class Announce
+    {
+        private bool Announced;
```
Comment (MyNick) [inline GameServer/Logic/GameObjects/Announce.cs]: CONVENTIONS
_announced
_eventTime
and so on

File: GameServer/Logic/GameObjects/Announce.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        public Announce(long eventTime, Announces id, bool isMapSpecific)
+        {
+            Announced = false;
+            EventTime = eventTime;
+            MessageId = id;
+            IsMapSpecific = isMapSpecific;
+        }
+
+        public bool IsAnnounced()
```
Comment (MyNick) [inline GameServer/Logic/GameObjects/Announce.cs]: please use property (shorter code = happy writers)

File: GameServer/Logic/GameObjects/Fountain.cs [hunk 3]
```diff
@@ -11,61 +11,51 @@ namespace LeagueSandbox.GameServer.Logic.GameObjects
...
-        protected const int NUM_SIDES = 2;// Change implementation when more game-modes with more teams are available
-        protected const float PERCENT_MAX_HEALTH_HEAL = 0.15f;
-        protected const float PERCENT_MAX_MANA_HEAL = 0.15f;
-        protected float fountainSize;
-        protected long healTickTimer;
-        protected List<Target> healLocations = new List<Target>();
+        private const float PERCENT_MAX_HEALTH_HEAL = 0.15f;
+        private const float PERCENT_MAX_MANA_HEAL = 0.15f;
+        private const float HEAL_FREQUENCY = 1000f;
+        private float X;
```
Comment (MyNick) [inline GameServer/Logic/GameObjects/Fountain.cs]: _x
_y...

File: GameServer/Logic/Maps/Map.cs [hunk 4]
```diff
@@ -25,7 +25,7 @@ public class Map
         protected long nextSpawnTime;
         protected long firstGoldTime; // Time that gold should begin to generate
         protected long nextSyncTime;
-        protected List<Pair<bool, Tuple<long, byte, bool>>> announcerEvents;
+        protected List<GameObjects.Announce> announcerEvents;
```
Comment (MyNick) [inline GameServer/Logic/Maps/Map.cs]: Great!

File: GameServer/Logic/Maps/Map.cs [hunk 5]
```diff
@@ -34,13 +34,15 @@ public class Map
         protected int id;
 
         protected CollisionHandler collisionHandler;
-        protected Fountain fountain;
+        protected Dictionary<TeamId, Fountain> Fountains;
```
Comment (MyNick) [inline GameServer/Logic/Maps/Map.cs]: _fountains

File: GameServer/Logic/Maps/Map.cs [hunk 6]
```diff
@@ -49,17 +51,22 @@ public Map(Game game, long firstSpawnTime, long spawnInterval, long firstGoldTim
             this.gameTime = 0;
             this.nextSpawnTime = firstSpawnTime;
             this.nextSyncTime = 10 * 1000;
-            this.announcerEvents = new List<Pair<bool, Tuple<long, byte, bool>>>();
+            this.announcerEvents = new List<GameObjects.Announce>();
```
Comment (MyNick) [inline GameServer/Logic/Maps/Map.cs]: if you use the underscore convention you own't need to use "this." all the time

File: GameServer/Logic/Maps/Map.cs [hunk 7]
```diff
@@ -164,23 +171,10 @@ public virtual void update(long diff)
...
-                    var isMapSpecific = i.Item2.Item3;
-
-                    if (gameTime >= eventTime)
-                    {
-                        PacketNotifier.notifyAnnounceEvent(messageId, isMapSpecific);
-                        i.Item1 = true;
-                    }
-                }
-            }
+            foreach (var announce in announcerEvents)
```
Comment (MyNick) [inline GameServer/Logic/Maps/Map.cs]: you should add {} to make sure everyone understand whats going on

File: GameServer/Logic/Maps/SummonersRift.cs [hunk 8]
```diff
@@ -96,30 +96,30 @@ public SummonersRift(Game game) : base(game, /*90*/5 * 1000, 30 * 1000, 90 * 100
...
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_OrderTurretShrine_A", -236.05f, -53.32f, 9999, 999, 0));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_ChaosTurretShrine_A", 14157.0f, 14456.0f, 9999, 999, 1));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T1_L_03_A", 574.66f, 10220.47f, 2550, 156, 0));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T1_L_02_A", 1106.26f, 6485.25f, 2550, 170, 0));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T1_C_06_A", 802.81f, 4052.36f, 2550, 190, 0));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T2_L_03_A", 3911.0f, 13654.0f, 2550, 156, 1));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T2_L_02_A", 7536.0f, 13190.0f, 2550, 170, 1));
-            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T2_L_01_A", 10261.0f, 13465.0f, 2550, 190, 1));
+            addObject(new Turret(this, Game.GetNewNetID(), "@Turret_T1_R_03_A", 10097.62f, 808.73f, 2550, 156, TeamId.TEAM_BLUE));
```
Comment (MyNick) [inline GameServer/Logic/Maps/SummonersRift.cs]: why don't we load it from a file?
what happen if we want to make a mode without turrets or something? it look kinda bad
feel free to disagree with it

File: GameServer/Logic/GameObjects/Announce.cs (hunk 2 above)
Comment (MythicManiac) [inline GameServer/Logic/GameObjects/Announce.cs]: Not to mention the private variable naming convention is wrong here, should be `_someVariable`, or just use properties like you suggested

File: GameServer/Logic/Maps/SummonersRift.cs (hunk 8 above)
Comment (MythicManiac) [inline GameServer/Logic/Maps/SummonersRift.cs]: This is going to be loaded from a while, but that's on my tasklist, not on @horato's. So this change is alright for now.

**Comment by MythicManiac** [comment]
I'm not guite sure what's up with that last codecov integration, but @horato you have some other changes you should make before this should be merged. So fix your code style and I'll merge this

**Comment by MythicManiac** [comment]
image

**Pull request merged**